overlord/ifacestate/handlers.go

```diff
@@ -1960,3 +1960,19 @@ func (m *InterfaceManager) doHotplugSeqWait(task *state.Task, _ *tomb.Tomb) erro
 	// no conflicting change for same hotplug key found
 	return nil
 }
+
+func (m *InterfaceManager) doRegenerateAllSecurityProfiles(task *state.Task, _ *tomb.Tomb) error {
+	// XXX: this will make snapd unusable for a long time :(
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	perfTimings := state.TimingsForTask(task)
+	defer perfTimings.Save(task.State())
+
+	if err := m.regenerateAllSecurityProfiles(perfTimings); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
just a nitpick, feel free to ignore
```suggestion
	return m.regenerateAllSecurityProfiles(perfTimings)
```